clang/lib/AST/Expr.cpp

```diff
@@ -1616,22 +1616,24 @@ QualType CallExpr::getCallReturnType(const ASTContext &Ctx) const {
   return FnType->getReturnType();
 }
 
-const Attr *CallExpr::getUnusedResultAttr(const ASTContext &Ctx) const {
+std::pair<const NamedDecl *, const Attr *>
+CallExpr::getUnusedResultAttr(const ASTContext &Ctx) const {
+  // If the callee is marked nodiscard, return that attribute
+  const Decl *D = getCalleeDecl();
```

Comment:
We should document this inversion in the release notes.This is more than just changing the name, it is preferring the `callee` decl here instead of the return type.  this is at least notable.

Reply:
Yes, that is definitely observable, as I presented in https://github.com/llvm/llvm-project/pull/112521#issuecomment-2417847257. This actually fixed an inconsistency bug between Clang and GCC/MSVC, with the latter two already always preferring the callee decl.

I will add a release note for this.